File: demo/viewjava/src/main/java/pers/shawxingkwok/kdatastore/demo/viewjava/MainFragment.java
package pers.shawxingkwok.kdatastore.demo.viewjava;

import android.annotation.SuppressLint;
import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import pers.shawxingkwok.kdatastore.demo.settings.Settings;
import pers.shawxingkwok.kdatastore.demo.viewjava.databinding.FragmentMainBinding;

public class MainFragment extends Fragment {
    public static MainFragment newInstance() {
        return new MainFragment();
    }
    private FragmentMainBinding binding;

    @Nullable
    @Override
    public View onCreateView(
        @NonNull LayoutInflater inflater,
        @Nullable ViewGroup container,
        @Nullable Bundle savedInstanceState
    ) {
        binding = FragmentMainBinding.inflate(getLayoutInflater(), container, false);
        return binding.getRoot();
    }

    @SuppressLint("NonConstantResourceId")
    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        // set the initial checked radioButton with `isDarkMode`
        if (Settings.isDarkMode().getValue() == null)
            binding.rbFollowSystem.setChecked(true);
        else if (Settings.isDarkMode().getValue())
            binding.rbDark.setChecked(true);
        else
            binding.rbLight.setChecked(true);

        // update `isDarkMode` in Listener
        binding.rgTheme.setOnCheckedChangeListener(
            (group, checkedId) -> {
                switch (checkedId){
                    case R.id.rb_followSystem:
                        Settings.isDarkMode().setValue(null);
                        break;

                    case R.id.rb_dark:
                        Settings.isDarkMode().setValue(true);
                        break;

                    case R.id.rb_light:
                        Settings.isDarkMode().setValue(false);
                        break;

                    default:
                        throw new IllegalStateException("Unexpected value: " + checkedId);
                }
            }
        );
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }
}

File: demo/viewkt/src/main/java/pers/shawxingkwok/kdatastore/demo/viewkt/MainFragment.kt
package pers.shawxingkwok.kdatastore.demo.viewkt

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import pers.shawxingkwok.kdatastore.demo.settings.Settings
import pers.shawxingkwok.kdatastore.demo.viewkt.databinding.FragmentMainBinding

class MainFragment : Fragment() {
    companion object {
        fun newInstance() = MainFragment()
    }

    private var _binding: FragmentMainBinding? = null
    private val binding get() = _binding!!

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?,
    ): View {
        _binding = FragmentMainBinding.inflate(layoutInflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        // set the initial checked radioButton with `isDarkMode`
        when (Settings.isDarkMode.value) {
            null -> binding.rbFollowSystem
            true -> binding.rbDark
            false -> binding.rbLight
        }
        .isChecked = true

        // update `isDarkMode` in Listener
        binding.rgTheme.setOnCheckedChangeListener { _, id ->
            Settings.isDarkMode.value =
                when(id){
                    R.id.rb_followSystem -> null
                    R.id.rb_dark -> true
                    R.id.rb_light -> false
                    else -> error("")
                }
        }
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }
}